File: ServerApp/Entities/Organisation.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using Everest.AspNetStartup.Binding;
using Everest.AspNetStartup.Models;
using Exam.Models.Statistics;
using Microsoft.AspNetCore.Mvc;
using Newtonsoft.Json;

namespace Exam.Entities
{
    [ModelBinder(BinderType = typeof(ItemValueModelBinder))]
    public class Organisation:Entity<long>
    {
        public string UserId { get; set; }
        public string AdminUserId { get; set; }

        public string Identifier { get; set; }
        public string Name { get; set; }
        public string Address { get; set; }

        public string ImageName { get; set; }
        public Uri ImageUrl { get; set; }
        
        [JsonIgnore]
        public virtual List<Admin> Admins { get; set; }
        public uint AdminCount { get; set; }

        [JsonIgnore]
        public virtual List<Examination> Examinations { get; set; }
        public uint ExaminationCount { get; set; }
        

        [JsonIgnore]
        public virtual List<Room> Rooms { get; set; }
        public uint RoomCount { get; set; }
        
        [NotMapped] public OrganisationStatistics Statistics { get; set; }
    }
}

File: ClientApp/src/models/entities/organisation.ts
import {Entity} from './entity';
import {User} from './user.entity';

export class Organisation extends Entity<number> {

  constructor(value?: any) {
    super();
    if (value) {
      this.id = value.id;
      this.name = value.name;
      this.address = value.address;
      this.identifier = value.identifier;

      this.statistics = value.statistics;

      this.userId = value.userId;
      this.user = value.user;

      this.adminUserId = value.adminUserId;
      this.adminUser = value.adminUser;
    }
  }

  userId: string;
  user: User;

  adminUserId: string;
  adminUser: User;

  name: string;
  identifier: string;

  address: string;

  userPrincipal: OrganisationUser = {};

  statistics: OrganisationStatistics = {};

  get url(): string {
    return `/organisations/${this.id}`;
  }

}


export class OrganisationStatistics {
  adminCount?: number = 0;
  examinationCount?: number = 0;
  waitingExaminationCount?: number = 0;
  progressExaminationCount?: number = 0;
  closedExaminationCount?: number = 0;
  roomCount?: number = 0;
  capacity?: number = 0;
}

export interface OrganisationUser {
  isAdmin?: boolean;
  isPrincipal?: boolean;
  userId?: string;
}
